.github/workflows/rust.yml

```diff
@@ -136,15 +147,32 @@ jobs:
           Remove-Item "${env:RUNNER_TEMP}\HIP-SDK-Installer.exe"
         if: runner.os == 'Windows' && steps.rocm-cache.outputs.cache-hit != 'true'
 
-      - name: Configure cache
+      - name: Configure source deps cache
         uses: actions/cache@1bd1e32a3bdc45362d1e726936510720a7c30a57 # v4.2.0
         with:
           path: |
             ~/.cargo/registry
             ~/.cargo/git
-          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.toml') }}
-          restore-keys: |
-            ${{ runner.os }}-cargo-
+            ~/.cargo/.crates2.json
+            ~/.cargo/.global-cache
```

Comment:
Why are these two files important?

Reply:
Thanks for checking this!
Actually, I think those files (and parts of the current `registry` and `git`) aren't needed.

In the docs, the paths to cache for crate sources are:
- registry/index
- registry/cache
- git/db

https://doc.rust-lang.org/cargo/guide/cargo-home.html#caching-the-cargo-home-in-ci

Caching registry/src and git/checkouts massively increases the number of files, and more than doubles the size. I'll re-check Windows after this change, because it might turn out to be faster with many fewer files.

`.global-cache` is a tiny marker file that stops backup software descending into the directory, but I'm pretty sure it gets re-created by cargo anyway.

Reply:
Those git checkouts are exactly the files that are used to build crates, that is why we cache them. Yes, they could be relatively large, especially as we upgrade dependencies over time and caches grow in size over time.

Reply:
Yes, but we only need to cache the compressed crate archives and bare git repository clones. Then once the cache is restored, cargo will uncompress/checkout the sources from those archives.

We definitely don't need to cache both the archives and the sources, that's more than double the size, and many more files.